public/app/plugins/datasource/elasticsearch/components/QueryEditor/RawQueryEditor.tsx

```diff
@@ -0,0 +1,106 @@
+import { css } from '@emotion/css';
+import { useCallback, useRef } from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { CodeEditor, Monaco, monacoTypes, useStyles2, Button, Stack } from '@grafana/ui';
+
+interface Props {
+  value?: string;
+  onChange: (value: string) => void;
+  onRunQuery: () => void;
+}
+
+export function RawQueryEditor({ value, onChange, onRunQuery }: Props) {
+  const styles = useStyles2(getStyles);
+  const editorRef = useRef<monacoTypes.editor.IStandaloneCodeEditor | null>(null);
+
+  const handleEditorDidMount = useCallback(
+    (editor: monacoTypes.editor.IStandaloneCodeEditor, monaco: Monaco) => {
+      editorRef.current = editor;
+
+      // Add keyboard shortcut for running query (Ctrl/Cmd+Enter)
+      editor.addCommand(monaco.KeyMod.CtrlCmd | monaco.KeyCode.Enter, () => {
+        onRunQuery();
+      });
+    },
+    [onRunQuery]
+  );
+
+  const handleFormat = useCallback(() => {
+    if (editorRef.current) {
+      editorRef.current.getAction('editor.action.formatDocument')?.run();
+    }
+  }, []);
+
+  const handleQueryChange = useCallback(
+    (newValue: string) => {
+      if (!newValue) {
+        return;
+      }
+      onChange(newValue);
+    },
+    [onChange]
+  );
+
+  return (
+    <div className={styles.container}>
+      <div className={styles.header}>
+        <Stack gap={1}>
+          <Button
+            size="sm"
+            variant="secondary"
+            icon="brackets-curly"
+            onClick={handleFormat}
+            tooltip="Format query (Shift+Alt+F)"
+          >
+            Format
+          </Button>
+          <Button size="sm" variant="primary" icon="play" onClick={onRunQuery} tooltip="Run query (Ctrl/Cmd+Enter)">
+            Run
+          </Button>
+        </Stack>
+      </div>
+      <CodeEditor
+        value={value ?? ''}
+        language="json"
+        height={200}
+        width="100%"
+        onBlur={handleQueryChange}
+        monacoOptions={{
```

Comment:
This is quite a dense object, is it worth pulling this out into a constant above the component?